msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CodesignTaskBase.cs

```diff
@@ -194,15 +241,107 @@ public override bool Execute ()
 				return true;
 
 			var codesignedFiles = new List<ITaskItem> ();
-			var resourcesToSign = Resources.Where (v => NeedsCodesign (v));
+			var resourcesToSign = Resources.Where (v => NeedsCodesign (v)).ToArray ();
```

Comment:
Do we really need the array conversion? IEnumerable of the foreach is good enough.